Character.cpp

```diff
@@ -86,32 +87,43 @@ int Character::takeDamage(int damage)
     return hitPoints;
 }
 
-
-#include <cassert>
 void Character::attackInternal(Character& other)
 {
     if( other.hitPoints <= 0 )
     {
         /*
-        When you defeat another Character: 
+        When you defeat another Character:
             a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;
+        defeatOpponent(*initialHitPoints, hitPoints);
+        defeatOpponent(*initialArmorLevel, armor);
+        defeatOpponent(*initialAttackDamage, attackDamage);
     }
 }
 
+void Character::defeatOpponent(int& initialValue, int& currentValue)
+{
+    if( currentValue < initialValue )
+    {
+        currentValue = initialValue;
+    }
+    currentValue *= 1.1f;
+    initialValue = currentValue;
+}
```

Comment:
This function does nothing with regard to defeating the opponent.
it resets/boosts the stats.
- [ ] name functions based on what they do
- [ ] revise